registry/storage/driver/walk.go

```diff
@@ -22,9 +22,14 @@ type WalkFn func(fileInfo FileInfo) error
 // to a directory, the directory will not be entered and Walk
 // will continue the traversal.  If fileInfo refers to a normal file, processing stops
 func WalkFallback(ctx context.Context, driver StorageDriver, from string, f WalkFn) error {
+	err, _ := doWalkFallback(ctx, driver, from, f)
+	return err
+}
+
+func doWalkFallback(ctx context.Context, driver StorageDriver, from string, f WalkFn) (error, bool) {
```

Comment:
Turns out the fallback implementation actually wasn't properly handling `If fileInfo refers to a normal file, processing stops` as found with a unit test. Added a bool return that can be used to gracefully stop the Walk.